pkg/fragment/definition.go

```diff
@@ -43,19 +62,44 @@ func WithTimingLabel(timingLabel string) DefinitionOption {
 	}
 }
 
-func (d *Definition) UrlWithParams(parameters url.Values) string {
+func (d *Definition) DynamicParts() []string {
+	return d.dynamicParts
+}
+
+func (d *Definition) UrlWithParams(parameters url.Values) *url.URL {
 	// This is already parsed before constructing the url in server.go, so we ignore errors
 	targetUrl, _ := url.Parse(d.Url)
 	targetUrl.RawQuery = parameters.Encode()
 
-	return targetUrl.String()
+	return targetUrl
 }
 
-func (d *Definition) IntoRequestable(params url.Values) multiplexer.Requestable {
+func (d *Definition) Requestable(target *url.URL, parameters map[string]string, query url.Values) (*Request, error) {
```

Comment:
```suggestion
func (d *Definition) Requestable(target *url.URL, pathParams map[string]string, query url.Values) (*Request, error) {
```

Other functions in this file refer to the query params as `parameters`. Hoping to disambiguate

Reply:
That's a lot more clear, thanks!